opensearch-project/OpenSearch-Dashboards
# [Workspace]Validate features parameter in workspace create and update API

### Description

This PR is for adding validate features field logic in workspace create and update API to align with the workspace create and update form. In current implementation, each workspace should have one use case.

### Issues Resolved



#7907

## Screenshot


No UI changes

## Testing the changes

### Prepare


- Clone branch code and run `yarn osd bootstrap --single-version ignore`
- Add below configs in `config/opensearch_dashboards.yml`
```
savedObjects.permission.enabled: true
workspace.enabled: true
uiSettings:
  overrides:
    'home:useNewHomePage': true
```
- Run `yarn start --no-base-path`
### Test with UI
- Login and visit workspace create page
- Click "Create workspace" button, the workspace should be created successfully
- Page will redirect to the workspace detail page
- Click edit button, add some description, click "Save changes", the workspace should be updated.
### Test features parameter validation
Can reference with unit test files: `src/plugins/workspace/server/routes/index.test.ts`

## Changelog

- feat: [Workspace]Validate features parameter in workspace create and update API

### Check List
